dig.go

```diff
@@ -954,36 +1022,52 @@ func isIgnoreUnexportedSet(f reflect.StructField) (bool, error) {
 	return allowed, err
 }
 
-// Checks that all direct dependencies of the provided param are present in
+// Checks that all direct dependencies of the provided parameters are present in
 // the container. Returns an error if not.
-func shallowCheckDependencies(c containerStore, p param) error {
+func shallowCheckDependencies(c containerStore, pl paramList) error {
 	var err errMissingTypes
 	var addMissingNodes []*dot.Param
-	walkParam(p, paramVisitorFunc(func(p param) bool {
-		ps, ok := p.(paramSingle)
-		if !ok {
-			return true
-		}
 
-		if ns := c.getValueProviders(ps.Name, ps.Type); len(ns) == 0 && !ps.Optional {
-			err = append(err, newErrMissingTypes(c, key{name: ps.Name, t: ps.Type})...)
-			addMissingNodes = append(addMissingNodes, ps.DotParam()...)
-		}
-
-		return true
-	}))
+	missingDeps := findMissingDependencies(c, pl.Params...)
+	for _, dep := range missingDeps {
+		err = append(err, newErrMissingTypes(c, key{name: dep.Name, t: dep.Type})...)
+		addMissingNodes = append(addMissingNodes, dep.DotParam()...)
+	}
 
 	if len(err) > 0 {
 		return err
 	}
 	return nil
 }
 
+func findMissingDependencies(c containerStore, params ...param) []paramSingle {
+	var missingDeps []paramSingle
+
+	for _, param := range params {
+		switch p := param.(type) {
+		case paramSingle:
+			if ns := c.getValueProviders(p.Name, p.Type); len(ns) == 0 && !p.Optional {
+				missingDeps = append(missingDeps, p)
+			}
+		case paramObject:
+			for _, f := range p.Fields {
+				missingDeps = append(missingDeps, findMissingDependencies(c, f.Param)...)
+
+			}
+		case paramList:
+			missingDeps = append(missingDeps, findMissingDependencies(c, p.Params...)...)
```

Comment:
This case is impossible. We can't have a paramList here because that represents the parameters of the constructor. There can't be another list of parameters inside it.